ODIN_II/SRC/partial_map.cpp

```diff
@@ -1134,14 +1134,32 @@ void instantiate_arithmetic_shift_right(nnode_t *node, short mark, netlist_t *ne
 	}
 	buf_node = make_1port_gate(BUF_NODE, width, width, node, mark);
 	/* connect inputs to outputs */
-	for(i = width - 2; i >= shift_size; i--)
+	for(i = width - 1; i >= shift_size; i--)
 	{
 		// connect higher input pin to lower output pin
 		remap_pin_to_new_node(node->input_pins[i], buf_node, i-shift_size);
 	}
-	/* connect first pin to outputs that don't have inputs connected */
-	i = width - 1;
-	remap_pin_to_new_node_range(node->input_pins[i], buf_node, i, i-shift_size);
+	
+	int pad_bit = width - 1 - shift_size;
+
+	/* connect ZERO to outputs that don't have inputs connected */
+	if(buf_node->input_pins[pad_bit]->type == INPUT)
+	{
+		for(i = width - 1; i >= width - shift_size; i--)
+		{
+			// connect 0 to lower outputs
+			add_input_pin_to_node(buf_node, copy_input_npin(buf_node->input_pins[pad_bit]), i);
+		}
+	}
+	else if(buf_node->input_pins[pad_bit]->type == OUTPUT)
```

Comment:
this should not be reached as we are only connecting the input ... this feels out of place.

Reply:
Possibly... This was put here so as to ensure that it would do the same thing as the now deleted function remap_pin_to_new_node_range. I'll try running it without that part there and see what happens

Reply:
I think the function was written to be generic but only ever used here, so I agree that it should be removed to simplify the logic since it's a very specific case here.

Reply:
This has now been changed